ww4ff/run/pretrain.py

```diff
@@ -0,0 +1,115 @@
+from ww4ff.utils.random import set_seed
+
+
+def expand(x):
+    return x.expand(3, -1, -1)
+
+
+def main():
+    def evaluate():
+        model.eval()
+        num_correct = 0
+        num_total = 0
+        for inputs, labels in tqdm(test_dl, total=len(test_dl)):
+            inputs = inputs.to(device)
+            labels = torch.tensor([x % 10 for x in labels.tolist()])
+            labels = labels.to(device)
+            scores = model(inputs, None)
+            num_correct += (scores.max(1)[1] == labels).float().sum().item()
+            num_total += scores.size(0)
+        logging.info(f'{num_correct / num_total}')
+        ws.increment_model(model, num_correct / num_total / 100)
+
+    apb = ArgumentParserBuilder()
+    apb.add_options(opt('--model', type=str, choices=model_names(), default='las'),
+                    opt('--workspace', type=str, default=str(Path('workspaces') / 'default')),
+                    opt('--load-weights', action='store_true'))
+    args = apb.parser.parse_args()
+
+    ws = Workspace(Path(args.workspace))
+    writer = ws.summary_writer
+    set_seed(SETTINGS.training.seed)
+
+    transform = transforms.Compose([transforms.RandomCrop(32, padding=4),
+                                    transforms.RandomHorizontalFlip(),
+                                    transforms.ToTensor(),
+                                    transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
+    test_transform = transforms.Compose([transforms.ToTensor(),
+                                         transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
+
+    trainset1 = torchvision.datasets.CIFAR10(root='./data', train=True, download=True, transform=transform)
+    testset1 = torchvision.datasets.CIFAR10(root='./data', train=False, download=True, transform=test_transform)
+
+    trainset2 = torchvision.datasets.CIFAR100(root='./data', train=True, download=True, transform=transform)
+    testset2 = torchvision.datasets.CIFAR100(root='./data', train=False, download=True, transform=test_transform)
+
+    transform = transforms.Compose([transforms.RandomCrop(32, padding=4),
+                                    transforms.RandomHorizontalFlip(),
+                                    transforms.ToTensor(),
+                                    transforms.Normalize((0.5,), (0.5,)),
+                                    expand])
+    test_transform = transforms.Compose([transforms.Pad((2, 2)),
+                                         transforms.ToTensor(),
+                                         transforms.Normalize((0.5,), (0.5,)),
+                                         expand])
+    trainset3 = torchvision.datasets.FashionMNIST(root='./data', train=True, download=True, transform=transform)
+    testset3 = torchvision.datasets.FashionMNIST(root='./data', train=False, download=True, transform=test_transform)
+    train_dl = tud.DataLoader(tud.ConcatDataset([trainset1, trainset2, trainset3]),
```

Comment:
isn't CIFAR100 has 100 classes? How can you concatenate a dataset of 10 classes with a dataset of 100 classes?

Reply:
I just did mod 10.